istioctl/pkg/waypoint/waypoint.go

```diff
@@ -87,6 +89,34 @@ func Cmd(ctx cli.Context) *cobra.Command {
 				}},
 			},
 		}
+		// Determine which traffic address type to apply the waypoint to, if none
+		// then default to "service" as the waypoint-for traffic address type.
+		validAddressTypes := map[string]bool{
+			"service":  true,
+			"workload": true,
+			"all":      true,
+			"none":     true,
+		}
+		if addressType != "" {
```

Comment:
Does the annotation not get set if the user passes it in?

Reply:
should be set to service when not otherwise set by the user afaik

Reply:
If the `addressType` == `""` then it will fall in to the default case of the switch statement. That's the intention at least.